bdb_transaction_cli/cli.py

```diff
@@ -1,15 +1,132 @@
-# -*- coding: utf-8 -*-
+import json
 
 import click
+from bigchaindb_common.crypto import generate_key_pair
+from bigchaindb_common.transaction import Transaction, \
+    Condition, Fulfillment, Metadata, Asset, Ed25519Fulfillment
 
+from bdb_transaction_cli.utils import json_argument, json_option, listify
 
-@click.command()
-def main(args=None):
-    """Console script for bdb_transaction_cli"""
-    click.echo("Replace this message by putting your code into "
-               "bdb_transaction_cli.cli.main")
-    click.echo("See click documentation at http://click.pocoo.org/")
+
+@click.group()
+def main():
+    pass
+
+
+@main.command()
+@click.option('--name', required=False, help=(
+    "Print the keys as shell variables,                             eg: "
+    "`export $(bdb generate_keys --name=bob)`"))
+def generate_keys(name):
+    """
+    Generate Ed25519 key pair.
+
+    Generates a random Ed25519 key pair separated by a space character.
+    First value is the public key, second is the private key.
+    """
+    priv, pub = generate_key_pair()
+    fmt = '{pub} {priv}'
+    if name:
+        fmt = '{name}_pub={pub} {name}_priv={priv}'
+    click.echo(fmt.format(name=name, pub=pub, priv=priv))
+
+
+@main.command()
+@click.argument('owner_after', required=True, nargs=-1)
+def generate_condition(owner_after):
+    """
+    Generate cryptoconditions from keys.
+
+    Generates a Ed25119 condition from a OWNER_AFTER or a ThresholdSha256
+    Condition from more than one OWNER_AFTER.
+    """
+    condition = Condition.generate(list(owner_after))
```

Comment:
`[]` and `list()` don't do the same thing, the first wraps, the second converts.

The conversion is because the `owners_after` argument is variadic and click is providing a tuple, and in `transactions.py` you have the following, so it breaks:

```python
         if isinstance(owners_after, tuple):         
             owners_after, threshold = owners_after  
         else:                                       
             threshold = len(owners_after)           
```